timeplus/client.go

```diff
@@ -283,44 +288,47 @@ func (s *TimeplusClient) queryStreamV2(sql string, batchCount int, batchBufferTi
 	config := utils.NewDefaultHTTPClientConfig()
 	res, err := utils.SSEHttpRequestWithAPIKey(http.MethodPost, createQueryUrl, query, config, s.apikey)
 	if err != nil {
-		return nil, nil, nil, fmt.Errorf("failed to create query : %w", err)
+		res.Body.Close()
```

Comment:
I'm not 100% sure (depends on your impl. of `SSEHttpRequestWithAPIKey`) you need to close it here or not. Just make sure it won't panic then it should be fine

Reply:
Yes, I think we dont need close the reponse here 
```go
res, err := client.Do(req)
if err != nil {
    return nil, err
}
defer res.Body.Close() // Make sure to close the response body
```